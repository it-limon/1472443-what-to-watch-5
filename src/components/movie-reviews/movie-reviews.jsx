import React, {useEffect} from "react";
import {getCommentsColumns} from "../../utils";
import {connect} from "react-redux";
import {getComments} from "../../store/selectors/data-selector";
import {loadCommentsList} from "../../store/api-actions";
import PropTypes from "prop-types";
import Props from "../../props";

const MovieReviews = (props) => {
  const {movie, comments, onLoadCommentsList} = props;

  useEffect(() => {
    onLoadCommentsList(movie.id);
  }, []);

  return (
    <div className="movie-card__reviews movie-card__row">
      {getCommentsColumns(comments).map((colIt, i) => (
        <div key={i} className="movie-card__reviews-col">
          {colIt.map((it) => (
            <div key={it.id} className="review">
              <blockquote className="review__quote">
                <p className="review__text">{it.comment}</p>

                <footer className="review__details">
                  <cite className="review__author">{it.user.name}</cite>
                  <time className="review__date" dateTime={it.date}>
                    {new Date(it.date).toLocaleDateString(`en-US`, {year: `numeric`, month: `long`, day: `numeric`})}
                  </time>
                </footer>
              </blockquote>

              <div className="review__rating">{it.rating.toFixed(1)}</div>
            </div>
          ))}
        </div>
      ))}
    </div>
  );
};

MovieReviews.propTypes = {
  movie: Props.movie,
  comments: PropTypes.arrayOf(Props.comment).isRequired,
  onLoadCommentsList: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  comments: getComments(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoadCommentsList(movieId) {
    dispatch(loadCommentsList(movieId));
  }
});

export {MovieReviews};
export default connect(mapStateToProps, mapDispatchToProps)(MovieReviews);
