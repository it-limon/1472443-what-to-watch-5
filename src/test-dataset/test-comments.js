export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];