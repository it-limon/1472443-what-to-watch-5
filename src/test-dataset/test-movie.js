export const testMovie = {
  id: 1,
  name: `El Camino: A Breaking Bad Movie`,
  posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
  previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
  backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
  backgroundColor: `#000000`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
  rating: 7.2,
  scoresCount: 36346,
  director: `Vince Gilligan`,
  starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
  runTime: 122,
  genre: `Drama`,
  released: 2019,
  isFavorite: true
};
