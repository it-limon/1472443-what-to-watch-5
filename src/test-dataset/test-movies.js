export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];