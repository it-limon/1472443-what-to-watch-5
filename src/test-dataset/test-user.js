export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};