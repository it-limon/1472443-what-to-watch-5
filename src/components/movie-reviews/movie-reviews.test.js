import React from "react";
import renderer from "react-test-renderer";
import {MovieReviews} from "./movie-reviews";
import {testMovie} from "../../test-dataset/test-movie";
import {testComments} from "../../test-dataset/test-comments";

const noop = () => {};

describe(`Render MovieReviews`, () => {
  
  it(`Render MovieReviews with 3 comments`, () => {
    const tree = renderer.create(
      <MovieReviews
        movie={testMovie}
        comments={testComments}
        onLoadCommentsList={noop}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieReviews with 0 comments`, () => {
    const tree = renderer.create(
      <MovieReviews
        movie={testMovie}
        comments={[]}
        onLoadCommentsList={noop}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
