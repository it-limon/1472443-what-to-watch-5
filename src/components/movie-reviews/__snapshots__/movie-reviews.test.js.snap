// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieReviews Render MovieReviews with 0 comments 1`] = `
<div
  className="movie-card__reviews movie-card__row"
/>
`;

exports[`Render MovieReviews Render MovieReviews with 3 comments 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 1
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            User 1
          </cite>
          <time
            className="review__date"
            dateTime="1995-12-17T03:24:00"
          >
            December 17, 1995
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.0
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 2
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            User 2
          </cite>
          <time
            className="review__date"
            dateTime="1996-12-17T03:24:00"
          >
            December 17, 1996
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        7.0
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 3
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            User 3
          </cite>
          <time
            className="review__date"
            dateTime="1997-12-17T03:24:00"
          >
            December 17, 1997
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.0
      </div>
    </div>
  </div>
</div>
`;
